### PR TITLE
New functionality for cross DC handling case if Kafka is not lossless #1322 (Somehow github decided to close my last PR)

### PR DESCRIPTION
**Pull request by wxing1292**

### CODE REVIEW

**Comment by wxing1292** [comment]
right now the replication info is still exposed in idl "shared", i can change it back to be in "history"
the functionality to enable / disable re-replication will be handled in the next PR, since this one is already too large

File: idl/github.com/uber/cadence/admin.thrift [hunk 1]
```diff
@@ -58,4 +70,21 @@ struct DescribeWorkflowExecutionResponse{
...
+  10: optional string domain
+  20: optional shared.WorkflowExecution execution
+  30: optional i64 (js.type = "Long") firstEventId
+  40: optional i64 (js.type = "Long") nextEventId
+  50: optional i32 maximumPageSize
+  60: optional binary nextPageToken
+}
+
+struct GetWorkflowExecutionRawHistoryResponse {
+  10: optional binary branchToken
```
Comment (samarabbas) [inline idl/github.com/uber/cadence/admin.thrift]: Can we remove this?

File: service/frontend/adminHandler.go [hunk 2]
```diff
@@ -118,21 +136,180 @@ func (adh *AdminHandler) DescribeWorkflowExecution(ctx context.Context, request
...
+			HistoryBatches:    []*gen.DataBlob{},
+			ReplicationInfo:   token.ReplicationInfo,
+			EventStoreVersion: common.Int32Ptr(token.EventStoreVersion),
+			NextPageToken:     nil, // no further pagination
+		}, nil
+	}
+
+	// TODO need to deal with transient decision if to be used by client getting history
+	var historyBatches []*gen.History
+	_, historyBatches, token.PersistenceToken, _, err = historyService.PaginateHistory(
```
Comment (samarabbas) [inline service/frontend/adminHandler.go]: Can you also emit history size metric with GetWorkflowExecutionRawHistory scope.

File: idl/github.com/uber/cadence/admin.thrift (hunk 1 above)
Comment (wxing1292) [reply idl/github.com/uber/cadence/admin.thrift]: sure

File: service/frontend/adminHandler.go (hunk 2 above)
Comment (wxing1292) [reply service/frontend/adminHandler.go]: sure

**Pull request merged**